src/main/java/com/github/philippheuer/credentialmanager/CredentialManager.java

```diff
@@ -186,6 +185,7 @@ public Optional<OAuth2Credential> getOAuth2CredentialByUserId(@NotNull String us
     @Synchronized
     public void load() {
         this.credentials = storageBackend.loadCredentials();
+        this.credentials.forEach(authenticationController::registerCredential);
```

Comment:
this is problematic because the identity providers wouldn't be registered yet... will do more refactoring